sdk/textanalytics/ai-text-analytics/src/healthResult.ts

```diff
@@ -66,3 +163,77 @@ export interface PagedHealthcareEntities extends PagedAsyncIterableHealthcareEnt
    */
   modelVersion: string;
 }
+
+/**
+ * Creates a user-friendly healthcare entity represented as a node in a graph
+ * @param entity - the healthcare entity returned by the service
+ * @internal
+ * @hidden
+ */
+function makeHealthcareEntitiesWithoutNeighbors(
+  entity: GeneratedHealthcareEntity
+): HealthcareEntity {
+  const { category, confidenceScore, isNegated, offset, text, links, subCategory } = entity;
+  return {
+    category,
+    confidenceScore,
+    isNegated,
+    offset,
+    text,
+    subCategory,
+    dataSource: links?.map(
+      ({ dataSource, id }): HealthcareEntityDataSource => ({ dataSource, dataSourceId: id })
+    ),
+    // initialize the neighbors map to be filled later.
+    relatedHealthcareEntities: new Map()
+  };
+}
+
+/**
+ * Connects input healthcare entities as a graph according to the relationship
+ * information the service sent.
+ * @param entities - healthcare entities represented as isolated nodes
+ * @param relations - relationship information between pairs of healthcare entities
+ *                  - using JSON pointers
+ * @internal
+ * @hidden
+ */
+function makeHealthcareEntitiesGraph(
+  entities: HealthcareEntity[],
+  relations: HealthcareRelation[]
+): HealthcareEntity[] {
+  for (const relation of relations) {
+    const relationType = relation.relationType;
+    const sourceIndex = parseHealthcareEntityIndex(relation.source);
+    const targetIndex = parseHealthcareEntityIndex(relation.target);
+    const sourceEntity = entities[sourceIndex];
+    const targetEntity = entities[targetIndex];
+    if (isHealthcareEntityRelationType(relationType)) {
+      sourceEntity.relatedHealthcareEntities.set(targetEntity, relationType);
+      if (relation.bidirectional) {
+        targetEntity.relatedHealthcareEntities.set(sourceEntity, relationType);
+      }
+    } else {
+      throw new Error(`Unrecognized healthcare entity relation type: ${relationType}`);
+    }
+  }
+  return entities;
+}
+
+/**
+ * Creates a healthcare entity in the convenience layer from the one sent by the service.
+ * @param document - incoming results sent by the service for a particular document
+ * @internal
+ * @hidden
+ */
+export function makeHealthcareEntitiesResult(
+  document: DocumentHealthcareEntities
+): HealthcareSuccessResult {
+  const { id, entities, relations, warnings, statistics } = document;
+  const newEntitiesWithoutNeighbors = entities.map(makeHealthcareEntitiesWithoutNeighbors);
+  const newEntities = makeHealthcareEntitiesGraph(newEntitiesWithoutNeighbors, relations);
```

Comment:
Got a little confused here. `makeHealthcareEntitiesGraph` is destructive to `newEntitiesWithoutNeighbors`, but returns the argument as well. This makes it look like a non-destructive functional program, but actually `newEntities` is triple-equal `newEntitiesWithoutNeighbors`.

Reply:
Fixed